resources/scripts/shared_functions.R

```diff
@@ -0,0 +1,115 @@
+      ylab("") +
+      scale_color_manual(values = c("#E0DD10", "#035C94")) +
+      coord_cartesian(clip = "off") +
+      theme(plot.margin = margin(1,1,1,1.1, "cm")) +
+      scale_x_reverse(limits = c(xmax,1), breaks = xmax:1, labels = xmax:1) +
+      annotation_custom(textGrob(textGrobMost, gp=gpar(fontsize=8, fontface = "bold")),xmin=-1,xmax=-1,ymin=-0.5,ymax=-0.5) +
+      annotation_custom(textGrob(textGrobLeast, gp=gpar(fontsize=8, fontface= "bold")),xmin=-xmax,xmax=-xmax,ymin=-0.5,ymax=-0.5)
+  )
+}
+
+prep_df_whichData <- function(subset_df, onAnVILDF = NULL){
+  subset_df %<>% separate(AccessWhichControlledData,
+                          c("WhichA", "WhichB", "WhichC", "WhichD", "WhichE", "WhichF", "WhichG", "WhichH", "WhichI", "WhichJ", "WhichK", "WhichM", "WhichN"),
+                          sep = ", ", fill="right") %>%
+    pivot_longer(starts_with("Which"),
+                 names_to = "WhichChoice",
+                 values_to = "whichControlledAccess") %>%
+    drop_na(whichControlledAccess) %>%
+    group_by(whichControlledAccess) %>%
+    summarize(count = n()) %>%
+    mutate(whichControlledAccess =
+             recode(whichControlledAccess,
+                    "All of Us*" = "All of Us",
+                    "UK Biobank*" = "UK Biobank",
+                    "Centers for Common Disease Genomics (CCDG)" = "CCDG",
+                    "The Centers for Mendelian Genomics (CMG)" = "CMG",
+                    "Clinical Sequencing Evidence-Generating Research (CSER)" = "CSER",
+                    "Electronic Medical Records and Genomics (eMERGE)" = "eMERGE",
+                    "Gabriella Miller Kids First (GMKF)" = "GMKF",
+                    "Genomics Research to Elucidate the Genetics of Rare Diseases (GREGoR)" = "GREGoR",
+                    "The Genotype-Tissue Expression Project (GTEx)" = "GTEx",
+                    "The Human Pangenome Reference Consortium (HPRC)" = "HPRC",
+                    "Population Architecture Using Genomics and Epidemiology (PAGE)" = "PAGE",
+                    "Undiagnosed Disease Network (UDN)" = "UDN",
+                    "Being able to pull other dbGap data as needed." = "Other",
+                    "Cancer omics datasets" = "Other",
+                    "GnomAD and ClinVar" = "None", #not controlled access
+             )
+    ) %>%
+    left_join(onAnVILDF, by="whichControlledAccess")
+
+  return(subset_df)
+}
+
+plot_which_data <- function(inputToPlotDF, subtitle = NULL){
+
+  toreturnplot <- ggplot(inputToPlotDF,
+                         aes(
+                           x = reorder(whichControlledAccess, -count),
+                           y = count,
+                           fill = AnVIL_Availability)
+                         ) +
+    geom_bar(stat="identity") +
+    theme_classic() +
+    theme(panel.background = element_blank(),
+          panel.grid = element_blank(),
+          axis.text.x = element_text(angle=45, hjust=1),
+          legend.position = "inside",
```

Comment:
```suggestion
          legend.position = c(0.8, 0.8)
```
I got an error `The 'legend.position.inside' theme element is not defined in the element hierarchy.` when running with arg `legend.position.inside` . I think you can specify the position like this, but I'd test to make sure.

Reply:
Hm, I know I had a struggle with this when I first tried to move the legends inside using only the numbers, and it told me what I was trying to use was deprecated. Could it be a version difference?

Reply:
Ok let's leave as is with a commented note in the function. I know the container uses a different version of R than I have so you're probably right. Does it work locally for you?

Reply:
Yes it does